prelude/toolchains/conan/defs.bzl

```diff
@@ -0,0 +1,791 @@
+        #"deps": attrs.list(attrs.dep(), default = []),
+        #"exported_deps": attrs.list(attrs.dep(), default = []),
+        #"exported_platform_deps": attrs.list(attrs.tuple(attrs.regex(), attrs.set(attrs.dep(), sorted = True)), default = []),
+        #"exported_preprocessor_flags": attrs.list(attrs.string(), default = []),
+        #"import_libs": attrs.dict(key = attrs.string(), value = attrs.source(), sorted = False, default = {}),
+        #"include_dirs": attrs.list(attrs.source(), default = []),
+        #"include_in_android_merge_map_output": attrs.bool(),
+        #"labels": attrs.list(attrs.string(), default = []),
+        #"licenses": attrs.list(attrs.source(), default = []),
+        #"provided_shared_libs": attrs.dict(key = attrs.string(), value = attrs.source(), sorted = False, default = {}),
+        #"shared_libs": attrs.dict(key = attrs.string(), value = attrs.source(), sorted = False, default = {}),
+        #"shared_link": attrs.list(attrs.string(), default = []),
+        #"static_libs": attrs.list(attrs.source(), default = []),
+        #"static_link": attrs.list(attrs.string(), default = []),
+        #"static_pic_libs": attrs.list(attrs.source(), default = []),
+        #"static_pic_link": attrs.list(attrs.string(), default = []),
+        #"supported_platforms_regex": attrs.option(attrs.regex(), default = None),
+        #"within_view": attrs.option(attrs.list(attrs.string())),
+
+def _conan_cxx_libraries_impl(ctx: "context") -> ["provider"]:
+    default_info = DefaultInfo(
+        default_outputs = ctx.attrs.main[DefaultInfo].default_outputs + flatten([c[DefaultInfo].default_outputs for c in ctx.attrs.components.values()]),
+        sub_targets = { n: c.providers for n, c in ctx.attrs.components.items() },
+    )
+    providers = [p for p in ctx.attrs.main.providers if type(p) != "DefaultInfo"]
+    providers.append(default_info)
+    return providers
+
+_conan_cxx_libraries = rule(
+    impl = _conan_cxx_libraries_impl,
+    attrs = {
+        "main": attrs.dep(doc = "The main package target, depends on all components."),
+        "components": attrs.dict(key = attrs.string(), value = attrs.dep(), doc = "The package's components."),
+    },
+    doc = "Helper rule to bundle Conan package components into a single target.",
+)
+
+def conan_dep(name: "string", components: {"string": "string"}, **kwargs):
+    """Bundle Conan package components into a single target.
+
+    The target itself represents the entire Conan package, including its
+    sub-components, if any. The individual components are exposed as sub-targets,
+    e.g. `:openssl` represents the entire openssl package, while `:openssl[crypto]`
+    represents only the `crypto` component.
+    """
+    native.cxx_library(
+        name = "_bundle_" + name,
+        exported_deps = components.values(),
+    )
+    _conan_cxx_libraries(
+        name = name,
+        main = ":_bundle_" + name,
+        components = components,
```

Comment:
I love we support this. Some context about components for others: https://docs.conan.io/en/latest/creating_packages/package_information.html#using-components